pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -342,7 +342,11 @@ func (r *ClusterInfoReconciler) getVendor(gitVersion string, isOpenShift bool) (
 	case strings.Contains(gitVersion, string(clusterv1beta1.KubeVendorICP)):
 		kubeVendor = clusterv1beta1.KubeVendorICP
 	case isOpenShift:
-		kubeVendor = clusterv1beta1.KubeVendorOpenShift
+		if r.isOpenShiftDedicated() {
```

Comment:
this looks strange. why not using a case before this case?

Reply:
done